### ISSUE TITLE
Stuck on blank screen during Conditional Access flow

### ISSUE THREAD

**Issue by Jim-Bar** [msintuneappsdk/ms-intune-app-sdk-ios#302]
Hi,

We are implementing the [Conditional Access flow](https://learn.microsoft.com/en-us/mem/intune/developer/app-sdk-ios#app-protection-ca-support-optional) in our application. When there is no Conditional Access, registering the user and receiving MAM policies works correctly. However when enabling it the app gets stuck on a blank screen between the steps "*Intune enrolls user (May trigger restart)*" and "*Is user compliant?*" (refer to the diagram in the link above).

Here is the order of execution:
1. We call `application.acquireToken(...)`
2. The Authenticator app is started
3. The completion block is invoked with error `MSALError.serverProtectionPoliciesRequired`
4. We call `IntuneMAMComplianceManager.instance().remediateCompliance(forIdentity: "upn@example.com", silent: true)`
5. The method `enrollmentRequest(...)` on `IntuneMAMEnrollmentDelegate` is invoked with status "Did succeed" (the SDK enrolled the user)
6. The SDK restarts the app
7. The app gets stuck and the method `identity(...)` on `IntuneMAMComplianceDelegate` never gets called
8. Killing then restarting the app works, but the delegate for compliance has still never been called

- iOS 14
- Intune SDK 17.1.2

Here is a video of the issue:

https://user-images.githubusercontent.com/8939266/214294636-3f3df272-2ca2-47b4-bab7-d73ab5c72705.mp4

Any idea of what could cause this?

**Comment by Jim-Bar** [comment]
Hi,

Adding up on this, we also reproduce the issue on other devices: iPhone 12 with iOS 16.2.

We see warnings in the logs, but we think those are unrelated to our problem (can you confirm?):

    [unspecified] container_create_or_lookup_app_group_path_by_app_group_identifier: client is not entitled
    [unspecified] container_create_or_lookup_app_group_path_by_app_group_identifier: client is not entitled
    [unspecified] A NULL string is not a valid group container identifier.
    [unspecified] container_create_or_lookup_app_group_path_by_app_group_identifier: client sent invalid parameters

We do not use an app group in the application.

**Comment by Jim-Bar** [comment]
Hi,

I am closing this issue because we finally sorted out the problem. There were two problems actually:

1. The Intune SDK was conflicting with our splashscreen manager at app startup (causing the blank screen)
2. The compliance delegate was registered too late (causing the delegate not to be called): it turns out the method `identity` is invoked by Intune after `application:willFinishLaunchingWithOptions:`, so this is a good place to register the delegate

We found about **2.** thanks to an old project [Wagr](https://github.com/msintuneappsdk/wagr/blob/main/Wagr/Application/Delegates/MAMComplianceDelegate.m).

I suggest adding the note about `application:willFinishLaunchingWithOptions:` in [the documentation](https://learn.microsoft.com/en-us/mem/intune/developer/app-sdk-ios#app-protection-ca-support-optional), so that others don’t run into the same problem.
